src/cargo/util/progress.rs

```diff
@@ -521,20 +520,20 @@ impl Format {
     fn render(&self, string: &mut String, msg: &str) {
         let mut avail_msg_len = self.max_width - string.len() - 15;
         let mut ellipsis_pos = 0;
-        if avail_msg_len <= 3 {
+        if avail_msg_len <= 1 {
             return;
         }
         for c in msg.chars() {
             let display_width = c.width().unwrap_or(0);
             if avail_msg_len >= display_width {
                 avail_msg_len -= display_width;
                 string.push(c);
-                if avail_msg_len >= 3 {
+                if avail_msg_len >= 1 {
                     ellipsis_pos = string.len();
                 }
             } else {
                 string.truncate(ellipsis_pos);
-                string.push_str("...");
+                string.push_str("…");
```

Comment:
We do not unconditionally output unicode characters but need to check either `shell().out_unicode()` or `shell().err_unicode()`.

Reply:
> Unicode is a superset of ascii. I assume you mean non-ascii Unicode?

Yes